### PR TITLE
[#6741] Update generators templates and VSIX to .Net 8

### PR DESCRIPTION
**Pull request by sw-joelmut**
Fixes # 6741
#minor

## Description
This PR updates all generators from `net6` to `net8`.

## Specific Changes
  - `Templates` 
    - Updates all related files where the `net6` configuration appear, changing it to `net8`.
    - Default template installation to `net8`, but leave `net6` and `net8` as supported options. Since we maintained both in the SDK.
  - `VSIX`
    - Updates all related files to target `net8`.
    - Fixed an issue when generating the `CoreBotWithTests` sample failed because it was referencing nonexistent files.

## Testing
The following images show all three bots working and the update reference in the VSIX samples.
imagen
imagen
imagen
imagen

### CODE REVIEW

File: generators/dotnet-templates/README.md [hunk 1]
```diff
@@ -256,7 +256,7 @@ Given the example above with `.nuspec` `version` of 4.22.0, running `nuget pack
 
 ```bash
 # install the locally built .nupkg (EchoBot template, assuming 4.9.0 version tag)
```
Comment (ceciliaavila) [inline generators/dotnet-templates/README.md:258]: Should this 4.9.0 version be changed?

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: generators/dotnet-templates/README.md (hunk 1 above)
Comment (sw-joelmut) [reply generators/dotnet-templates/README.md]: Applied, thanks!

**Review by ceciliaavila — APPROVED** [review]

**Review by JhontSouth — APPROVED** [review]

**Comment by sw-joelmut** [comment]
✅

File: generators/dotnet-templates/Microsoft.BotFramework.CSharp.CoreBot/content/CoreBot/README.md [hunk 2]
```diff
@@ -68,7 +68,7 @@ Once you created the LUIS model, update `appsettings.json` with your `LuisAppId`
 
 [Bot Framework Emulator](https://github.com/microsoft/botframework-emulator) is a desktop application that allows bot developers to test and debug their bots on localhost or running remotely through a tunnel.
 
-- Install the Bot Framework Emulator version 4.9.0 or greater from [here](https://github.com/Microsoft/BotFramework-Emulator/releases)
+- Install the Bot Framework Emulator version 4.22.0 or greater from [here](https://github.com/Microsoft/BotFramework-Emulator/releases)
```
Comment (andres-robinet-sw) [inline generators/dotnet-templates/Microsoft.BotFramework.CSharp.CoreBot/content/CoreBot/README.md:71]: Latest version for emulator is `4.14.x`. I would just go for "Install the latest v4 Bot Framework Emulator" and use `latest` for version. See [Linking to releases](https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases)

File: generators/dotnet-templates/Microsoft.BotFramework.CSharp.EchoBot/content/README.md [hunk 3]
```diff
@@ -43,7 +43,7 @@ This bot has been created using [Bot Framework](https://dev.botframework.com), i
 
 [Bot Framework Emulator](https://github.com/microsoft/botframework-emulator) is a desktop application that allows bot developers to test and debug their bots on localhost or running remotely through a tunnel.
 
-- Install the Bot Framework Emulator version 4.9.0 or greater from [here](https://github.com/Microsoft/BotFramework-Emulator/releases)
+- Install the Bot Framework Emulator version 4.22.0 or greater from [here](https://github.com/Microsoft/BotFramework-Emulator/releases)
```
Comment (andres-robinet-sw) [inline generators/dotnet-templates/Microsoft.BotFramework.CSharp.EchoBot/content/README.md:46]: Latest version for emulator is `4.14.x`. I would just go for "Install the latest v4 Bot Framework Emulator" and use `latest` for version. See [Linking to releases](https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases)

File: generators/dotnet-templates/Microsoft.BotFramework.CSharp.EmptyBot/content/README.md [hunk 4]
```diff
@@ -43,7 +43,7 @@ This bot has been created using [Bot Framework](https://dev.botframework.com), i
 
 [Bot Framework Emulator](https://github.com/microsoft/botframework-emulator) is a desktop application that allows bot developers to test and debug their bots on localhost or running remotely through a tunnel.
 
-- Install the Bot Framework Emulator version 4.9.0 or greater from [here](https://github.com/Microsoft/BotFramework-Emulator/releases)
+- Install the Bot Framework Emulator version 4.22.0 or greater from [here](https://github.com/Microsoft/BotFramework-Emulator/releases)
```
Comment (andres-robinet-sw) [inline generators/dotnet-templates/Microsoft.BotFramework.CSharp.EmptyBot/content/README.md:46]: Latest version for emulator is `4.14.x`. I would just go for "Install the latest v4 Bot Framework Emulator" and use `latest` for version. See [Linking to releases](https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases)

File: generators/dotnet-templates/README.md [hunk 5]
```diff
@@ -58,7 +58,7 @@ The Empty Bot template is the minimal skeleton code for a bot.  It provides a st
 
 
 ## Installation
-1.  Install [.NET Core SDK][4] version 3.1 or higher
+1.  Install [.NET Core SDK][4] version 8.0 or higher
```
Comment (andres-robinet-sw) [inline generators/dotnet-templates/README.md:61]: .NET Core SDK link is "[60]" not "[4]". Check the link references at the end of the document.

Comment (JhontSouth) [reply generators/dotnet-templates/README.md]: fixed

File: generators/dotnet-templates/Microsoft.BotFramework.CSharp.EmptyBot/content/README.md (hunk 4 above)
Comment (JhontSouth) [reply generators/dotnet-templates/Microsoft.BotFramework.CSharp.EmptyBot/content/README.md]: fixed

File: generators/dotnet-templates/Microsoft.BotFramework.CSharp.EchoBot/content/README.md (hunk 3 above)
Comment (JhontSouth) [reply generators/dotnet-templates/Microsoft.BotFramework.CSharp.EchoBot/content/README.md]: fixed

File: generators/dotnet-templates/Microsoft.BotFramework.CSharp.CoreBot/content/CoreBot/README.md (hunk 2 above)
Comment (JhontSouth) [reply generators/dotnet-templates/Microsoft.BotFramework.CSharp.CoreBot/content/CoreBot/README.md]: fixed

**Pull request closed without merging**